src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -0,0 +1,389 @@
+using System.Xml;
+using System.Net;
+
+namespace Microsoft.DotNet.Build.Tasks.Feed
+{
+    sealed class BlobFeedAction: IFeedAction<string>
+    {
+        private MSBuild.TaskLoggingHelper Log;
+        private static readonly CancellationTokenSource TokenSource = new CancellationTokenSource();
+        private static readonly CancellationToken CancellationToken = TokenSource.Token;
+        private static readonly string DownloadDirectory = Path.Combine(Directory.GetCurrentDirectory(), "download");
+        private static readonly string IndexDirectory = Path.Combine(Directory.GetCurrentDirectory(), "tmp");
+
+        public BlobFeed feed;
+        public int MaxClients { get; set; } = 8;
+
+        public BlobFeedAction(string accountName, string accountKey, string containerName, MSBuild.TaskLoggingHelper Log)
+        {
+            this.feed = new BlobFeed(accountName, accountKey, containerName, Log);
+            this.Log = Log;
+        }
+
+        public Task<bool> DeleteFromFeed(string filterString = "")
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<IEnumerable<string>> ListPackagesOnFeed(string filterString = "")
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<IEnumerable<string>> ListVersionsOfPackage(string packageName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<bool> PullFromFeed(string filterString = "")
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task<bool> PushToFeed(IEnumerable<string> items, string relativePath = "", bool containerExists = false)
+        {            
+            if (containerExists)
+            {
+                if (Directory.Exists(DownloadDirectory))
+                {
+                    Directory.Delete(DownloadDirectory, true);
+                }
+                Directory.CreateDirectory(DownloadDirectory);
+                using (var clientThrottle = new SemaphoreSlim(this.MaxClients, this.MaxClients))
+                {
+                    List<string> listAzureBlobs = await ListAzureBlobs(feed.AccountName, feed.AccountKey, feed.ContainerName);
+                    List<string> indexJsonBlobs = listAzureBlobs.FindAll(blob => blob.Contains("index.json"));
+
+                    // Merge index.json involves downloading the existing feed's index.jsons
+                    await Task.WhenAll(items.Select(item => MergeIndexes(CancellationToken, item, relativePath, indexJsonBlobs)));
+                }
+            }
+            if (CancellationToken.IsCancellationRequested)
+            {
+                Log.LogError("Task UploadToAzure cancelled");
+                CancellationToken.ThrowIfCancellationRequested();
+            }
+
+            List<string> indexList = new List<string>(Directory.EnumerateFiles(Path.Combine(IndexDirectory, relativePath), "index.json", SearchOption.AllDirectories));
```

Comment:
You are duplicating the file path here and below that should be shared to prevent getting out of sync.

Reply:
fixed.